### ISSUE TITLE
Error on creating cluster while starting new node on existing database

### ISSUE THREAD

**Issue by mohamnag** [orientechnologies/orientdb#5338]
I'm desperately trying to upgrade my v2.0.15 clustered DB to v2.1.5 however I'm running into funny situation here.
As suggested to me, I shutdown all the clusters' nodes and now I'm trying to start new ones using the existing database files. However I'm constantly getting following error even when using the node name of a previously existing node:
```
com.orientechnologies.orient.server.distributed.ODistributedException: Error on creating cluster 'payment-option_b3d5c6d2096e' in class 'payment-option'
	at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.installLocalClusterPerClass(OHazelcastPlugin.java:1611)
	at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.installDbClustersForLocalNode(OHazelcastPlugin.java:1280)
	at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.loadDistributedDatabases(OHazelcastPlugin.java:1421)
	at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.startup(OHazelcastPlugin.java:187)
	at com.orientechnologies.orient.server.OServer.registerPlugins(OServer.java:979)
	at com.orientechnologies.orient.server.OServer.activate(OServer.java:346)
	at com.orientechnologies.orient.server.OServerMain.main(OServerMain.java:41)
Caused by: com.orientechnologies.orient.server.distributed.ODistributedException: Quorum cannot be reached because there are no nodes available
	at com.orientechnologies.orient.server.hazelcast.OHazelcastDistributedDatabase.calculateQuorum(OHazelcastDistributedDatabase.java:364)
	at com.orientechnologies.orient.server.hazelcast.OHazelcastDistributedDatabase.send2Nodes(OHazelcastDistributedDatabase.java:142)
	at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.sendRequest(OHazelcastPlugin.java:363)
	at com.orientechnologies.orient.server.distributed.ODistributedStorage.command(ODistributedStorage.java:317)
	at com.orientechnologies.orient.server.distributed.ODistributedStorage.addCluster(ODistributedStorage.java:1275)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.addCluster(ODatabaseDocumentTx.java:1257)
	at com.orientechnologies.orient.core.metadata.schema.OClassImpl.createClusterIfNeeded(OClassImpl.java:2081)
	at com.orientechnologies.orient.core.metadata.schema.OClassImpl.addCluster(OClassImpl.java:1058)
	at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.installLocalClusterPerClass(OHazelcastPlugin.java:1604)
```

my write quorum is set to 1 to avoid any need but it seems it is simply impossible to start any node because the quorum will never be reached anyway.

**Comment by mohamnag** [comment]
there is only one log as this is the first node that should start correctly. situation is:
 - the database directory is remaining from a v2.0.15 and is copied with no change to v2.1.5
 - the node name set for this first node is new

following the startup log, where you see `....` I removed name of DB classes:
```
           .                                          
          .`        `                                 
          ,      `:.                                  
         `,`    ,:`                                   
         .,.   :,,                                    
         .,,  ,,,                                     
    .    .,.:::::  ````                                 :::::::::     :::::::::   
    ,`   .::,,,,::.,,,,,,`;;                      .:    ::::::::::    :::    :::  
    `,.  ::,,,,,,,:.,,.`  `                       .:    :::      :::  :::     ::: 
     ,,:,:,,,,,,,,::.   `        `         ``     .:    :::      :::  :::     ::: 
      ,,:.,,,,,,,,,: `::, ,,   ::,::`   : :,::`  ::::   :::      :::  :::    :::  
       ,:,,,,,,,,,,::,:   ,,  :.    :   ::    :   .:    :::      :::  :::::::     
        :,,,,,,,,,,:,::   ,,  :      :  :     :   .:    :::      :::  :::::::::   
  `     :,,,,,,,,,,:,::,  ,, .::::::::  :     :   .:    :::      :::  :::     ::: 
  `,...,,:,,,,,,,,,: .:,. ,, ,,         :     :   .:    :::      :::  :::     ::: 
    .,,,,::,,,,,,,:  `: , ,,  :     `   :     :   .:    :::      :::  :::     ::: 
      ...,::,,,,::.. `:  .,,  :,    :   :     :   .:    :::::::::::   :::     ::: 
           ,::::,,,. `:   ,,   :::::    :     :   .:    :::::::::     ::::::::::  
           ,,:` `,,.                                  
          ,,,    .,`                                  
         ,,.     `,                                          GRAPH DATABASE  
       ``        `.                                                          
                 ``                                          orientdb.com
                 `                                    

2015-11-18 09:00:06:162 INFO  Loading configuration from: /opt/orientdb/config/orientdb-server-config.xml... [OServerConfigurationLoaderXml]
2015-11-18 09:00:06:495 WARNI Assigning distributed node name: sslc-db-1 [OHazelcastPlugin]
2015-11-18 09:00:08:456 WARNI [sslc-db-1] opening database 'sslc'... [OHazelcastPlugin]
2015-11-18 09:00:08:557 WARNI [sslc-db-1] updated distributed configuration for database: sslc:
----------
{
  "version": 0,
  "autoDeploy": true,
  "hotAlignment": false,
  "executionMode": "undefined",
  "readQuorum": 1,
  "writeQuorum": 1,
  "failureAvailableNodesLessQuorum": false,
  "readYourWrites": true,"clusters":{
    "internal": {
},
    "index": {
},
    "*": {
  "servers": ["sslc-db-backup-1","sslc-db-1","<NEW_NODE>"]
},
    "oschedule_node1444062165793": {
  "servers": ["sslc-db-backup-1","sslc-db-1","<NEW_NODE>"]
},
....

    "_studio_node1444062165793": {
  "servers": ["sslc-db-backup-1","sslc-db-1","<NEW_NODE>"]
},
....
    "user_node1444661341242": {
  "servers": ["sslc-db-backup-1","sslc-db-1","<NEW_NODE>"]
},
....
    }
}
---------- [OHazelcastPlugin]
2015-11-18 09:00:08:596 WARNI [sslc-db-1] updated distributed configuration for database: sslc:
----------
{
  "version": 0,
  "autoDeploy": true,
  "hotAlignment": false,
  "executionMode": "undefined",
  "readQuorum": 1,
  "writeQuorum": 1,
  "failureAvailableNodesLessQuorum": false,
  "readYourWrites": true,"clusters":{
    "internal": {
},
    "index": {
},
    "*": {
  "servers": ["sslc-db-backup-1","sslc-db-1","<NEW_NODE>"]
},
    "oschedule_node1444062165793": {
  "servers": ["sslc-db-backup-1","sslc-db-1","<NEW_NODE>"]
},
....
    "_studio_node1444062165793": {
  "servers": ["sslc-db-backup-1","sslc-db-1","<NEW_NODE>"]
},
....
    "ouser_node1444661341242": {
  "servers": ["sslc-db-backup-1","sslc-db-1","<NEW_NODE>"]
},
....
    }
}
---------- [OHazelcastPlugin]
2015-11-18 09:00:09:322 SEVER [sslc-db-1] error on creating cluster 'oschedule_sslc-db-1' in class 'OSchedule':  [OHazelcastPlugin][sslc-db-1] Error on starting distributed plugin
com.orientechnologies.orient.server.distributed.ODistributedException: Error on creating cluster 'oschedule_sslc-db-1' in class 'OSchedule'
    at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.installLocalClusterPerClass(OHazelcastPlugin.java:1611)
    at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.installDbClustersForLocalNode(OHazelcastPlugin.java:1280)
    at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.loadDistributedDatabases(OHazelcastPlugin.java:1421)
    at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.startup(OHazelcastPlugin.java:187)
    at com.orientechnologies.orient.server.OServer.registerPlugins(OServer.java:979)
    at com.orientechnologies.orient.server.OServer.activate(OServer.java:346)
    at com.orientechnologies.orient.server.OServerMain.main(OServerMain.java:41)
Caused by: com.orientechnologies.orient.server.distributed.ODistributedException: Quorum cannot be reached because there are no nodes available
    at com.orientechnologies.orient.server.hazelcast.OHazelcastDistributedDatabase.calculateQuorum(OHazelcastDistributedDatabase.java:364)
    at com.orientechnologies.orient.server.hazelcast.OHazelcastDistributedDatabase.send2Nodes(OHazelcastDistributedDatabase.java:142)
    at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.sendRequest(OHazelcastPlugin.java:363)
    at com.orientechnologies.orient.server.distributed.ODistributedStorage.command(ODistributedStorage.java:317)
    at com.orientechnologies.orient.server.distributed.ODistributedStorage.addCluster(ODistributedStorage.java:1275)
    at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.addCluster(ODatabaseDocumentTx.java:1257)
    at com.orientechnologies.orient.core.metadata.schema.OClassImpl.createClusterIfNeeded(OClassImpl.java:2081)
    at com.orientechnologies.orient.core.metadata.schema.OClassImpl.addCluster(OClassImpl.java:1058)
    at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.installLocalClusterPerClass(OHazelcastPlugin.java:1604)
    ... 6 more
```

**Comment by mohamnag** [comment]
using v2.0.15, using a new node name does not make any trouble and server starts successfully. also on v2.1.5 if I set node name to something used before (for example using v2.0.15) the problem does not appear.

**Comment by lvca** [comment]
So with v2.1.5 can you confirm this is fixed?

**Comment by mohamnag** [comment]
I'm not sure what you mean with fixed, here again an overview:
v2.0.15, using new node name and existing database directory ~> server starts successfully and creates new clusters for new node name

v2.1.5, using node name that was used by v2.0.15 and existing database directory ~> server starts successfully

v2.1.5, using NEW node name and existing database directory ~> server fails to start, it obviously tries to create new clusters but fails.

the last case does not seem a functional situation, it means first node in a cluster will always fail to start with a new node name because it can not reach the quorum (why a lonely node has to reach quorum?) and it can only be started with a node name which has been previously created by an older version.

**Comment by mohamnag** [comment]
this is getting even more problematic, I have now new exceptions even using existing node names when database is running on v2.1.5. When I add new classes to database I get following exception:
```
catching com.orientechnologies.orient.server.distributed.ODistributedException: Error on creating cluster 'new-class-name_node1447768640786' in class 'new-class-name'
	at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.installClustersOfClass(OHazelcastPlugin.java:1230) ~[?:?]
	at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.propagateSchemaChanges(OHazelcastPlugin.java:1162) ~[?:?]
	at com.orientechnologies.orient.server.distributed.ODistributedStorage.command(ODistributedStorage.java:320) ~[orientdb-server-2.1.5.jar:2.1.5]
	at com.orientechnologies.orient.core.command.OCommandRequestTextAbstract.execute(OCommandRequestTextAbstract.java:63) ~[orientdb-core-2.1.5.jar:2.1.5]
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.command(ONetworkProtocolBinary.java:1319) ~[orientdb-server-2.1.5.jar:2.1.5]
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:396) ~[orientdb-server-2.1.5.jar:2.1.5]
	at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:223) ~[orientdb-server-2.1.5.jar:2.1.5]
	at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:77) ~[orientdb-core-2.1.5.jar:2.1.5]
Caused by: com.orientechnologies.orient.server.distributed.ODistributedException: Quorum cannot be reached because there are no nodes available
	at com.orientechnologies.orient.server.hazelcast.OHazelcastDistributedDatabase.calculateQuorum(OHazelcastDistributedDatabase.java:364) ~[?:?]
	at com.orientechnologies.orient.server.hazelcast.OHazelcastDistributedDatabase.send2Nodes(OHazelcastDistributedDatabase.java:142) ~[?:?]
	at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.sendRequest(OHazelcastPlugin.java:363) ~[?:?]
	at com.orientechnologies.orient.server.distributed.ODistributedStorage.command(ODistributedStorage.java:317) ~[orientdb-server-2.1.5.jar:2.1.5]
	at com.orientechnologies.orient.server.distributed.ODistributedStorage.addCluster(ODistributedStorage.java:1275) ~[orientdb-server-2.1.5.jar:2.1.5]
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.addCluster(ODatabaseDocumentTx.java:1257) ~[orientdb-core-2.1.5.jar:2.1.5]
	at com.orientechnologies.orient.core.metadata.schema.OClassImpl.createClusterIfNeeded(OClassImpl.java:2081) ~[orientdb-core-2.1.5.jar:2.1.5]
	at com.orientechnologies.orient.core.metadata.schema.OClassImpl.addCluster(OClassImpl.java:1058) ~[orientdb-core-2.1.5.jar:2.1.5]
	at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.installClustersOfClass(OHazelcastPlugin.java:1223) ~[?:?]
	... 7 more

```

this is pretty critical and means I have to either turn off clustering completely and use single instances or to roll back to v2.0.15.

**Comment by lvca** [comment]
I tried your use case and works. Could I have your database to try with it locally? In case send to support - at - orientdb.com. Thanks.

**Comment by mohamnag** [comment]
I will provide you the DB on given email, I wanted to give another details which may be important to be aware of, I'm using this docker image to run my DB: github.com/mohamnag/ecs-orientdb
currently I switched to single instance DB with disabled clustering which is not optimal but it at least works.

**Comment by lvca** [comment]
The problem seems in the node names: you have configured 4 nodes, but no node is active to distributed the request. You should remove the node names from distributed-config.json under your database directory first.

**Comment by mohamnag** [comment]
I can confirm that by removing all the other node names I could start the first node in distributed mode successfully. however I'm still unable to run the second node due to #5340 to be able to confirm that distributed mode works as expected.

PS: was this on the migration guide that one shall remove node names manually?

**Comment by lvca** [comment]
Commented and closed #5340. Please look at my comment. Could you now retry if you can start nodes?
